### PR TITLE
Synopsys-Operator can support multiple versions of Alert

### PR DESCRIPTION
**Pull request by mphammer**

### CODE REVIEW

File: pkg/apps/alert/latest/versions.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+		"blackduck-cfssl": "1.0.0",
+	},
+	"4.0.0": {
+		"blackduck-alert": "4.0.0",
+		"blackduck-cfssl": "1.0.0",
+	},
+}
+
+// getImageTag returns the url for an image
+func (c *Creater) getImageTag(version, name string) string {
```
Comment (jdartigalongue) [inline pkg/apps/alert/latest/versions.go]: OK but you don't actually use this function to get the container tags. 
The CRD should probably be modified to have a version field.

Comment (mphammer) [reply pkg/apps/alert/latest/versions.go]: I removed the imageTag function and added a Version field to Alert

File: pkg/apps/alert/alert.go [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+	var retryWait time.Duration = 10
+	for i := 0.0; i <= attempts; i++ {
+		ns, err := util.GetNamespace(a.kubeClient, namespace)
+		log.Debugf("Namespace: %v, status: %v", namespace, ns.Status)
+		time.Sleep(retryWait * time.Second)
+		if err != nil {
+			log.Infof("Deleted the namespace %+v", namespace)
+			break
+		}
+		if i >= 10.0 {
```
[the remark this answers is not in the document]
Comment (mphammer) [reply pkg/apps/alert/alert.go]: Good point. I restructured that code to print less

File: pkg/apps/alert/alert.go [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+	if err != nil {
+		return fmt.Errorf("unable to find the namespace %+v due to %+v", namespace, err)
+	}
+	// Delete the namespace
+	err = util.DeleteNamespace(a.kubeClient, namespace)
+	if err != nil {
+		return fmt.Errorf("unable to delete the namespace %+v due to %+v", namespace, err)
+	}
+	// Verify whether the namespace deleted
+	var attempts = 30.0
```
[the remark this answers is not in the document]
Comment (mphammer) [reply pkg/apps/alert/alert.go]: Changed to int

File: pkg/alert/crdhandler.go [hunk 4]
```diff
@@ -44,6 +46,20 @@ type HandlerInterface interface {
...
 
+// State contains the state of the OpsSight
+type State string
+
+const (
+	// Running is used when the instance is running
+	Running State = "Running"
+	// Stopped is used when the instance is about to stip
+	Stopped State = "Stopped"
+	// Updating is used when the instance is about to updating
```
[the remark this answers is not in the document]
Comment (mphammer) [reply pkg/alert/crdhandler.go]: Fixed this in Alert pkg and Blackduck pkg

File: pkg/alert/crdhandler.go [hunk 5]
```diff
@@ -44,6 +46,20 @@ type HandlerInterface interface {
 	ObjectUpdated(objOld, objNew interface{})
 }
 
+// State contains the state of the OpsSight
+type State string
+
+const (
+	// Running is used when the instance is running
+	Running State = "Running"
+	// Stopped is used when the instance is about to stip
```
[the remark this answers is not in the document]
Comment (mphammer) [reply pkg/alert/crdhandler.go]: Fixed this in Alert pkg and Blackduck pkg

**Pull request merged**